atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+            self.logger.warning('No positions available in H5MD file. Other particle attributes will not be stored')
+            return self._system_info
+
+        def get_value(value, steps, path=''):
+            if value is None:
+                return value
+            if isinstance(value, h5py.Group):
+                path_value = f'{path}.value' if path else 'value'
+                value = self._data_parser.get(path_value)
+                path_step = f'{path}.step' if path else 'step'
+                attr_steps = self._data_parser.get(path_step)
+                if value is None or attr_steps is None:
+                    self.logger.warning('Missing values or steps in particle attributes. These attributes will not be stored.')
+                    return None
+                elif attr_steps.sort() != steps.sort() if attr_steps is not None else False:
+                    self.logger.warning('Distinct trajectory lengths of particle attributes not supported. These attributes will not be stored.')
+                    return None
+                else:
+                    return value
+            else:
+                return [value] * n_frames
+
+        # get the steps based on the positions
+        steps = self._data_parser.get(f'{self._path_group_positions_all}.step')
+        if steps is None:
+            self.logger.warning('No step information available in H5MD file. System information cannot be parsed.')
+            return self._system_info
+        self.trajectory_steps = steps
+
+        # get the rest of the particle quantities
+        values_dict = {'system': {}, 'calculation': {}}
+        times = self._data_parser.get(f'{self._path_group_positions_all}.time')
+        values_dict['system']['time'] = times
+        values_dict['calculation']['time'] = times
+        values_dict['system']['positions'] = positions
+        values_dict['system']['n_atoms'] = self._n_atoms
+        system_keys = {'labels': 'system', 'velocities': 'system', 'forces': 'calculation'}
+        for key, sec_key in system_keys.items():
+            path = f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}'
+            value = self._data_parser.get(path)
+            values_dict[sec_key][key] = get_value(value, steps, path=path)
+
+        # get the box quantities
+        box = self._data_parser.get(f'{self._path_group_particles_all}.box')
+        if box is not None:
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
+            for box_key, sec_key in box_attributes.items():
+                path = f'{self._path_group_particles_all}.box.{self._nomad_to_box_group_map[box_key]}'
+                value = self._data_parser.get(path, isattr=True)
+                values_dict[sec_key][box_key] = [value] * n_frames if value is not None else None
+            box_keys = {'lattice_vectors': 'system'}
```

Comment:
also here, and will we add more keys in box_keys?

Reply:
Here I even removed the loop.

Reply:
yes thanks.